File: app/src/main/java/com/andysklyarov/finnotify/framework/SoapCurrencyInRubDataSource.java
package com.andysklyarov.finnotify.framework;

import com.andysklyarov.finnotify.data.CurrencyInRubDataSource;
import com.andysklyarov.finnotify.domain.CurrencyInRub;
import com.andysklyarov.finnotify.framework.soap.SoapCbrUtils;

import java.time.LocalDate;

import io.reactivex.Single;


public class SoapCurrencyInRubDataSource implements CurrencyInRubDataSource {

    @Override
    public Single<CurrencyInRub> getCurrencyOnDate(String currencyCode, LocalDate date) {
        return SoapCbrUtils.getCurrencyOnDate(currencyCode, date);
    }

    @Override
    public Single<LocalDate> getLastDate() {
        return SoapCbrUtils.getLastServerDate();
    }
}

File: app/src/main/java/com/andysklyarov/finnotify/framework/SoapCurrencyInRubDataSource.kt
package com.andysklyarov.finnotify.framework

import com.andysklyarov.finnotify.data.CurrencyInRubDataSource
import com.andysklyarov.finnotify.domain.CurrencyInRub
import com.andysklyarov.finnotify.framework.soap.SoapCbrUtils
import io.reactivex.Single
import java.time.LocalDate

class SoapCurrencyInRubDataSource : CurrencyInRubDataSource {
    override fun getCurrencyOnDate(currencyCode: String, date: LocalDate): Single<CurrencyInRub> {
        return SoapCbrUtils.getCurrencyOnDate(currencyCode, date)
    }

    override fun getLastDate(): Single<LocalDate> {
        return SoapCbrUtils.getLastServerDate()
    }
}